eliot/_message.py

```diff
@@ -106,16 +124,57 @@ def write(self, logger=None, action=None):
         """
         if logger is None:
             logger = _output._DEFAULT_LOGGER
-        if action is None:
-            action = currentAction()
-        if action is None:
-            action = _defaultAction
-        contents = self._contents.copy()
-        contents["timestamp"] = self._timestamp()
-        if action is not None:
-            contents["task_uuid"] = action._identification["task_uuid"]
-            contents["task_level"] = action._nextTaskLevel().level
-        logger.write(contents, self._serializer)
+        logged = self._freeze(action=action)
+        logger.write(logged.as_dict(), self._serializer)
```

Comment:
As a first pass it seems like changing return from `Message.write` is unnecessary, and just another bit of extra work, so let's omit it if you don't feel it's necessary.